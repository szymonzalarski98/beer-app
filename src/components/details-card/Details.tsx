import React, { useCallback } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { colors, fontSize, deviceWidth } from '../../styles/helpers.style';
import theme from '../../styles/theme.style';
import Property from '../Property';

interface Props {
  name: string;
  description: string;
  abv: number;
  ibu: number;
  ebc: number;
  ph: number;
};

const NUM_OF_LINES = 5;

const Details: React.SFC<Props> = (props) => {
  const { name, description, abv, ibu, ebc, ph } = props;
  const [showMoreButton, setShowMoreButton] = React.useState(false);
  const [showMore, setShowMore] = React.useState(false);
  const onTextLayout = React.useCallback((e) => {
    setShowMoreButton(e.nativeEvent.lines.length >= NUM_OF_LINES);
  }, []);
  return (
    <View style={styles.container}>
      {
        showMore
        ?
        <>
          <Text style={theme.title}>{name}</Text>
          <Text style={theme.text}>{description}</Text>
          {showMore && <Text style={[theme.text, styles.label]} onPress={() => setShowMore(!showMore)}>Hide</Text>}
        </>
        :
        <>
          <Text style={theme.title}>{name}</Text>
          <Text
            style={theme.text}
            numberOfLines={NUM_OF_LINES}
            onTextLayout={onTextLayout}
          >
            {description}
          </Text>
          {showMoreButton && <Text style={[theme.text, styles.label]} onPress={() => setShowMore(!showMore)}>Show more</Text>}
          <Property title="Abv: " text={abv} />
          <Property title="Ibu: " text={ibu} />
          <Property title="Ebc: " text={ebc} />
          <Property title="pH: " text={ph} />
        </>
      }
    </View>
  );
};

export default Details;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  label: {
    fontWeight: '700',
    color: '#000',
    marginVertical: 5,
  }
});