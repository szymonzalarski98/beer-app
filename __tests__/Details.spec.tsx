import React from 'react';
import { render } from 'react-native-testing-library';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Details from '../src/components/details-card/Details';

const mockStore = configureMockStore();

describe('Test Details component', () => {
  it ('Should renders correctly', () => {
    const store = mockStore({
      name: 'test',
      description: 'test',
      abv: 1,
      ibu: 1,
      ebc: 1,
      ph: 1,
    });
    const { baseElement, getByText } = render(
      <Provider store={store}>
        <Details name="Test" description="Test description" abv={10} ibu={100} ebc={30} ph={20} />
      </Provider>
    );
    expect(getByText('Test')).not.toBeNull();
    expect(getByText('Test description')).not.toBeNull();
    expect(getByText('10')).not.toBeNull();
    expect(getByText('100')).not.toBeNull();
    expect(getByText('30')).not.toBeNull();
    expect(getByText('20')).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  });
})