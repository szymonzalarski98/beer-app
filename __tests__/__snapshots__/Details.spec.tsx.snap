// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Details component Should renders correctly 1`] = `undefined`;
